flang/lib/Optimizer/OpenMP/MapInfoFinalization.cpp

```diff
@@ -486,6 +490,157 @@ class MapInfoFinalizationPass
       // iterations from previous function scopes.
       localBoxAllocas.clear();
 
+      // First, walk `omp.map.info` ops to see if any record members should be
+      // implicitly mapped.
+      func->walk([&](mlir::omp::MapInfoOp op) {
+        mlir::Type underlyingType =
+            fir::unwrapRefType(op.getVarPtr().getType());
+
+        if (!fir::isRecordWithAllocatableMember(underlyingType))
```

Comment:
I guess this won't work for all cases, as when `underlyingType` is an array of records, for instance.
It could be a TODO.